include/range/v3/utility/iterator.hpp

```diff
@@ -726,80 +742,150 @@ namespace ranges
             return reverse_iterator<I>(i);
         }
 
-        /// \cond
-        namespace detail
+        template<typename I>
+        struct move_iterator
         {
-            template<typename I>
-            struct move_cursor
+        private:
+            CONCEPT_ASSERT(InputIterator<I>());
+            I current_ = detail::value_init{};
```

Comment:
`InputIterator` does not require convertibility from `value_init_t`; this could explode.